test/unit/models/classes/LtiProvider/ConfigurableLtiProviderRepositoryTest.php

```diff
@@ -98,25 +81,41 @@ public function testSearchByOauthKey()
 
     public function testConstructorWithNullProviderListThrowsException()
     {
-        $subject = new ConfigurableLtiProviderRepository([
-            ConfigurableLtiProviderRepository::OPTION_LTI_PROVIDER_LIST => null
-        ]);
+        $subject = $this->createSubject(null);
+
         $this->expectException(InvalidArgumentException::class);
 
         $subject->count();
     }
 
     public function testConstructorWithInvalidProviderListThrowsException()
     {
-        $subject = new ConfigurableLtiProviderRepository([
-            ConfigurableLtiProviderRepository::OPTION_LTI_PROVIDER_LIST => json_decode(
-                file_get_contents(__DIR__ . '/_resources/incomplete_lti_provider_list.json'),
-                true
-            )
-        ]);
+        $subject = $this->createSubject(__DIR__ . '/_resources/incomplete_lti_provider_list.json');
+
         $this->expectException(InvalidArgumentException::class);
         $this->expectExceptionMessage('Missing key \'callback_url\' in LTI provider list.');
 
         $subject->count();
     }
+
+    private function createSubject(string $providerListPath = null)
+    {
+        $subject = new ConfigurableLtiProviderRepository(
+            [
+                ConfigurableLtiProviderRepository::OPTION_LTI_PROVIDER_LIST => $providerListPath ? json_decode(
+                    file_get_contents($providerListPath),
```

Comment:
should we use something else like flysystem inserted and set up a test folder instead

Reply:
or at least move here the path / dir and just provide the name here

Reply:
This was done previously, I just created a single method to avoid repeat 5x: https://github.com/oat-sa/extension-tao-lti/pull/270/files#diff-36de226ea64dce6b79ca91bad8ab7da4L64

Reply:
let's also add an improvement inside the improvement page , because i think this should not be the final solution